Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/securitycontextconstraints/provider.go

```diff
@@ -83,13 +84,16 @@ func NewSimpleProvider(scc *api.SecurityContextConstraints) (SecurityContextCons
 // container's security context then it will not be changed.  Validation should be used after
 // the context is created to ensure it complies with the required restrictions.
 //
-// WARNING: this method will change values on the container's security context.  If that is
-// undesirable (ie in the case of trying to generate and validate against multiple SCCs )
-// then a copy should be passed.
+// NOTE: this method works on a copy of the SC of the container.  It is up to the caller to apply
+// the SC if validation passes.
 func (s *simpleProvider) CreateSecurityContext(pod *api.Pod, container *api.Container) (*api.SecurityContext, error) {
 	var sc *api.SecurityContext = nil
 	if container.SecurityContext != nil {
-		sc = container.SecurityContext
+		scCopy, err := conversion.DeepCopy(container.SecurityContext)
```

Comment:
shouldn't need a deep copy here... you're only setting top level fields. should be able to do something like this: http://play.golang.org/p/YKmv6NEZU5
```
var sc api.SecurityContext
if container.SecurityContext != nil {
  sc = *container.SecurityContext
}
...
return &sc, nil
```